sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfigurationTest.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.attributes.SemanticAttributes;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+class OpenTelemetrySdkAutoConfigurationTest {
+
+  @Test
+  void resourcePrioritizesUser() {
+    Resource resource =
+        OpenTelemetrySdkAutoConfiguration.configureResource(
+            ConfigProperties.createForTest(
+                Collections.singletonMap("otel.resource.attributes", "telemetry.sdk.name=test")));
+    assertThat(resource.getAttributes().get(SemanticAttributes.TELEMETRY_SDK_NAME))
```

Comment:
I didn't even realize that this was a thing that end-users could do. Is this desirable behavior?

Reply:
I think we'll need it for service name? I don't think a user should do something bad but it seems to make sense semantically to prioritize them if they do go to the effort.

Reply:
I don't know if it's generally desirable or not, but it certainly is the way it works today, so have a test for it is good. :)